packages/ipfs/src/core/components/dag/resolve.js

```diff
@@ -1,16 +1,45 @@
 'use strict'
 
+const CID = require('cids')
 const { parseArgs } = require('./utils')
 const { withTimeoutOption } = require('../../utils')
 
 module.exports = ({ ipld, preload }) => {
-  return withTimeoutOption(async function * resolve (cid, path, options) { // eslint-disable-line require-await
+  return withTimeoutOption(async function resolve (cid, path, options) { // eslint-disable-line require-await
     [cid, path, options] = parseArgs(cid, path, options)
 
     if (options.preload !== false) {
       preload(cid)
     }
+    const { signal, timeout } = options
 
-    yield * ipld.resolve(cid, path, { signal: options.signal })
+    try {
+      let lastCid = cid
+      let lastRemainderPath = path
+
+      if (path) {
+        for await (const { value, remainderPath } of ipld.resolve(cid, path, {
+          signal,
+          timeout
```

Comment:
`ipld.resolve` does not support a `timeout` option.

Reply:
Gotcha! Missed looking up this [doc](https://github.com/ipld/js-ipld#resolvecid-path-options). Removed the `timeout` option.